services/app-api/src/postgres/contractAndRates/submitContract.ts

```diff
@@ -3,70 +3,68 @@ import type { ContractType } from '../../domain-models/contractAndRates'
 
-type SubmitContractArgsType = {
+export type SubmitContractArgsType = {
     contractID: string // revision ID
     submittedByUserID: UpdateInfoType['updatedBy']
     submittedReason: UpdateInfoType['updatedReason']
 }
 // Update the given revision
 // * invalidate relationships of previous revision by marking as outdated
 // * set the UpdateInfo
-async function submitContract(
+export async function submitContract(
     client: PrismaClient,
     args: SubmitContractArgsType
 ): Promise<ContractType | NotFoundError | Error> {
     const { contractID, submittedByUserID, submittedReason } = args
     const currentDateTime = new Date()
 
     try {
-        // Find current contract revision with related rates
-        // query only the submitted revisions on the related rates
-        return await client.$transaction(async (tx) => {
-            const currentRev = await tx.contractRevisionTable.findFirst({
-                where: {
-                    contractID: contractID,
-                    submitInfoID: null,
-                },
-                include: {
-                    draftRates: {
-                        include: includeLatestSubmittedRateRev,
-                    },
+        // find the current contract with related rates
+        const currentRev = await client.contractRevisionTable.findFirst({
+            where: {
+                contractID: contractID,
+                submitInfoID: null,
+            },
+            include: {
+                draftRates: {
+                    include: includeDraftRates,
                 },
-            })
+            },
+        })
 
-            if (!currentRev) {
-                const err = `PRISMA ERROR: Cannot find the current rev to submit with contract id: ${contractID}`
-                console.error(err)
-                return new NotFoundError(err)
-            }
+        if (!currentRev) {
+            const err = `PRISMA ERROR: Cannot find the current rev to submit with contract id: ${contractID}`
+            console.error(err)
+            return new NotFoundError(err)
+        }
 
-            // Given related rates, confirm rates valid by submitted by checking for revisions
-            // If rates have no revisions, we know it is invalid and can throw error
-            const relatedRateRevs = currentRev.draftRates.map(
-                (c) => c.revisions[0]
-            )
-            const everyRelatedRateIsSubmitted = relatedRateRevs.every(
-                (rev) => rev !== undefined
-            )
-            if (!everyRelatedRateIsSubmitted) {
-                const message =
-                    'Attempted to submit a contract related to a rate that has not been submitted.'
-                console.error(message)
-                return new Error(message)
-            }
+        // get the related rate revisions and any unsubmitted rates
+        const relatedRateRevs = currentRev.draftRates.map((c) => c.revisions[0])
```

Comment:
Technically, we should only be submitting rates that are parented by this contract. This is another thing that will change once we have unlock-resubmit rate, in which case you should only be able to submit DRAFT rates with this contract.

Reply:
Ok I wrote another test where I put together a contract with two rates, call `submitRate `on one of the rates , and then call `submitContract`, comparing the `submitInfos` between them. 

I filter out for `draftRates` that don't already have a `submitInfo`, which I took to mean it hasn't been submitted yet and are in DRAFT still. Let me know if that's not right though!